### ISSUE TITLE
`Unexpected "..."` when assigning tuple with variable length to type alias

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#11102]
```py
Foo = tuple[int, ...] # error: Unexpected "..."
```
https://mypy-play.net/?mypy=latest&python=3.9&flags=strict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=79574764450b7fd74e512b1ca6bd844a

related: #11098

**Comment by Zeckie** [comment]
This is valid:
```python
from typing import Tuple
Foo = Tuple[int, ...]
a: Foo = (1 ,2, 3)

reveal_type(a) # note: Revealed type is "builtins.tuple[builtins.int]"
```

Is it that mypy can't tell whether `tuple` is being used as a type or a function?
```python
from typing import Tuple
reveal_type(Tuple[int]) # note: Revealed type is "builtins.object"
reveal_type(tuple[int]) # note: Revealed type is "def (iterable: typing.Iterable[builtins.int*] =) -> builtins.tuple[builtins.int*]" 
```

**Comment by sobolevn** [comment]
Happens on `master` as well. I will send a PR with the fix.

**Comment by hauntsaninja** [comment]
I believe this was fixed (a while back) on master #10766
